Sources/SwiftSyntax/Raw/RawSyntax.swift

```diff
@@ -526,6 +526,9 @@ extension RawSyntax {
     initializingLeadingTriviaWith: (UnsafeMutableBufferPointer<RawTriviaPiece>) -> Void,
     initializingTrailingTriviaWith: (UnsafeMutableBufferPointer<RawTriviaPiece>) -> Void
   ) -> RawSyntax {
+    if let defaultText = kind.defaultText {
+      assert(text.isEmpty || kind.defaultText == text)
+    }
```

Comment:
How would you feel about `assert(kind.defaultText == nil || text.isEmpty || kind.defaultText == text)`?

Reply:
Much better